docs/proposals/scheduling/20220510-load-aware-scheduling.md

```diff
@@ -0,0 +1,137 @@
+
+#### Story 4
+
+The koordlet may not be able to report the latest resource usage due to node exceptions. Such nodes should be avoided during scheduling to prevent unexpected exceptions.
+
+### Notes/Constraints
+
+### Implementation Details
+
+![image](/docs/images/load-aware-scheduling-arch.svg)
+
+The scheduling plugin filters abnormal nodes and scores them according to resource usage. This scheduling plugin extends the Filter/Score/Reserve/Unreserve extension points defined in the Kubernetes scheduling framework.
+
+#### Filter unhealthy nodes
+
+By default, abnormal nodes are filtered, and users can decide whether to enable or not by configuring as needed.
+
+- Filter nodes where koordlet fails to update NodeMetric. If the configuration enables, the plugin will exclude nodes with *nodeMetrics.status.updateTime >= nodeMetricUpdateMaxWindowInSeconds*
+- Filter nodes by utilization thresholds. If the configuration enables, the plugin will exclude nodes with *latestUsageUtilization >= utilizationThreshold*. In the filtering phase, only the resource utilization is obtained from the latest NodeMetric, and the resource usage of the allocated but not yet counted Pods does not participate in the calculation, so as to allocate resources to the newly created Pods and avoid scheduling failure due to unreasonable estimates
+
+
+#### Score algorithm
+
+The core logic of the scoring algorithm is to select the node with the smallest resource usage. However, considering the delay of resource usage reporting and the delay of Pod startup time, the resource requests of the Pods that have been scheduled and the Pods currently being scheduled within the time window will also be estimated, and the estimated values will be involved in the calculation.
+
+The following is the algorithm:
+1. Get the total amount of resources of the current node as *T*
+2. Get the resource usage of the current node from NodeMetric as *A*
+3. Estimate the resource usage of the current Pod as *B*
+    1. If Limit is declared in PodSpec, then *B = Limit*
+    2. else *B = Request * scalingFactor*
+    3. If Request and Limit are not defined, then *B* = *defaultResourceRequest(250m)*
+4. According to the algorithm in step 3, estimate the resource usage of the Pod that has been allocated by the current node within the NodeMetric update window as *C*
+5. Estimate the total resource usage of the current node as *P*, *P = A + B + S*
+6. Calculate the score
+    1. If *P >= T*, then *score = 0*
+    2. else *score = (cpu((T - P) * 100 / T * cpuWeight) + memory((T - P) * 100 / T * memoryWeight)) / weightSum*
```

Comment:
I can get the point: the score is the weighted average of the estimated free rate of each resource.
But the formula is still unclear. e.g. `cpu((T - P) * 100 / T * cpuWeight) + memory()`. Can you please revise the formulation?

Reply:
Are you confused T/P these variables are per resource?